pinot-common/src/main/java/com/linkedin/pinot/common/protocols/SegmentFinalProtocol.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.common.protocols;
+
+import com.alibaba.fastjson.JSONObject;
+
+/*
+ * TODO Add unit tests for this after we finalize the protocol elements.
+ * TODO Finalize the protocol elements.
+ *
+ * This class encapsulates the segment finalization protocol used by the server and the controller for
```

Comment:
I would change finalization to something else to avoid reader confusion with Java finalization.